### PR TITLE
DOCS-2082: Add camera encoder gantry snippets

### PR DESCRIPTION
**Pull request by skyleilani**
Labeled as a draft while I review and test code snippets

### CODE REVIEW

File: components/gantry/gantry.go [hunk 1]
```diff
@@ -48,16 +48,40 @@ type Gantry interface {
...
+	//    // Get the current positions of the axes of the gantry in millimeters.
+	//    position, err := myGantry.Position(context.Background(), nil)
 	Position(ctx context.Context, extra map[string]interface{}) ([]float64, error)
 
 	// MoveToPosition is in meters
 	// This will block until done or a new operation cancels this one
+	//
+	//    myGantry, err := gantry.FromRobot(robot, "my_gantry")
+	//
+	//    // Create a list of positions for the axes of the gantry to move to. Assume in this example that the gantry is multi-axis, with 3 axes.
```
Comment (npentrel) [inline components/gantry/gantry.go:63]: [Test](https://github.com/viamrobotics/rdk/actions/runs/8743490939/job/23994141269?pr=3824) complains that this line is too long. 
```suggestion
	//    // Create a list of positions for the axes of the gantry to move to.
	//    // Assume in this example that the gantry is multi-axis, with 3 axes.
```

**Review by npentrel — APPROVED** [review]
LGTM once tests are passing

**Comment by npentrel** [comment]
@skyleilani I think you need to sign the contributor license agreement (see top comment).
@njooma who should review these PRs?

**Comment by njooma** [comment]
@npentrel anyone on SDK could review these.

@skyleilani can you post a screenshot of what these comments look like when rendered?

**Review by purplenicole730 — APPROVED** [review]

**Comment by npentrel** [comment]
@njooma happy with the screenshots? Let us know so we can merge

**Review by njooma — APPROVED** [review]

**Pull request merged**